### ISSUE TITLE
Wrong versions if package changed versioning scheme

### ISSUE THREAD

**Issue by kindlycat** [google/osv.dev#1215]
For example python package "neutron" have changed versioning scheme from calver to semver.
https://osv.dev/vulnerability/GHSA-4pmp-38hf-rmwj
New versions marked as affected, but this vulnerability fixed a long time ago.

**Comment by oliverchang** [comment]
Hi @kindlycat,

Thanks for reporting.

Hmm, we sort according to PyPI's version sorting rules, and it's unfortunate that the "newer" semver versions sort before the calver versions. That said, I don't actually see any calver versions on https://pypi.org/project/neutron/#history, so either they got deleted, or this advisory is completely wrong. 

I've created https://github.com/github/advisory-database/pull/2062 to get this fixed upstream on GitHub.

**Comment by andrewpollock** [comment]
Looks like https://github.com/github/advisory-database/pull/2062 resulted in a withdrawal of the offending record (although you'd be forgiven for not realising that, because the web interface doesn't make that evident at all)

```
curl "https://api.osv.dev/v1/vulns/GHSA-4pmp-38hf-rmwj" | jq .

{
  "id": "GHSA-4pmp-38hf-rmwj",
  "summary": "OpenStack Neutron allows remote authenticated users to cause a denial of service",
  "details": "OpenStack Neutron before 2013.2.4, 2014.x before 2014.1.2, and Juno before Juno-2 allows remote authenticated users to cause a denial of service (crash or long firewall rule updates) by creating a large number of allowed address pairs.",
  "aliases": [
    "CVE-2014-3555"
  ],
  "modified": "2023-04-14T18:48:12.698227Z",
  "published": "2022-05-17T03:06:07Z",
  "withdrawn": "2023-04-14T18:33:38Z",
  "database_specific": {
    "cwe_ids": [],
    "severity": "MODERATE",
    "github_reviewed": true,
    "github_reviewed_at": "2023-02-08T19:55:29Z",
    "nvd_published_at": "2014-07-23T14:55:00Z"
  },
  "references": [
    {
      "type": "ADVISORY",
      "url": "https://nvd.nist.gov/vuln/detail/CVE-2014-3555"
    },
    {
      "type": "WEB",
      "url": "https://access.redhat.com/errata/RHSA-2014:1078"
    },
    {
      "type": "WEB",
      "url": "https://access.redhat.com/errata/RHSA-2014:1119"
    },
    {
      "type": "WEB",
      "url": "https://access.redhat.com/errata/RHSA-2014:1120"
    },
    {
      "type": "WEB",
      "url": "https://access.redhat.com/security/cve/CVE-2014-3555"
    },
    {
      "type": "WEB",
      "url": "https://bugs.launchpad.net/neutron/+bug/1336207"
    },
    {
      "type": "WEB",
      "url": "https://bugzilla.redhat.com/show_bug.cgi?id=1118833"
    },
    {
      "type": "PACKAGE",
      "url": "https://opendev.org/openstack/neutron"
    },
    {
      "type": "WEB",
      "url": "https://web.archive.org/web/20200228142429/http://www.securityfocus.com/bid/68765"
    },
    {
      "type": "WEB",
      "url": "http://lists.openstack.org/pipermail/openstack-announce/2014-July/000255.html"
    },
    {
      "type": "WEB",
      "url": "http://rhn.redhat.com/errata/RHSA-2014-1119.html"
    },
    {
      "type": "WEB",
      "url": "http://rhn.redhat.com/errata/RHSA-2014-1120.html"
    },
    {
      "type": "WEB",
      "url": "http://seclists.org/oss-sec/2014/q3/200"
    }
  ],
  "affected": [
    {
      "package": {
        "name": "neutron",
        "ecosystem": "PyPI",
        "purl": "pkg:pypi/neutron"
      },
      "ranges": [
        {
          "type": "ECOSYSTEM",
          "events": [
            {
              "introduced": "0"
            },
            {
              "fixed": "2013.2.4"
            }
          ]
        }
      ],
      "versions": [
        "0.0",
        "10.0.5",
        "10.0.6",
        "10.0.7",
        "11.0.3",
        "11.0.4",
        "11.0.5",
        "11.0.6",
        "11.0.7",
        "11.0.8",
        "12.0.0",
        "12.0.0.0b3",
        "12.0.0.0rc1",
        "12.0.0.0rc2",
        "12.0.1",
        "12.0.2",
        "12.0.3",
        "12.0.4",
        "12.0.5",
        "12.0.6",
        "12.1.0",
        "12.1.1",
        "13.0.0",
        "13.0.0.0b1",
        "13.0.0.0b2",
        "13.0.0.0b3",
        "13.0.0.0rc1",
        "13.0.0.0rc2",
        "13.0.1",
        "13.0.2",
        "13.0.3",
        "13.0.4",
        "13.0.5",
        "13.0.6",
        "13.0.7",
        "14.0.0",
        "14.0.0.0b1",
        "14.0.0.0b2",
        "14.0.0.0b3",
        "14.0.0.0rc1",
        "14.0.1",
        "14.0.2",
        "14.0.3",
        "14.0.4",
        "14.1.0",
        "14.2.0",
        "14.3.0",
        "14.3.1",
        "14.4.0",
        "14.4.1",
        "14.4.2",
        "15.0.0",
        "15.0.0.0b1",
        "15.0.0.0rc1",
        "15.0.0.0rc2",
        "15.0.1",
        "15.0.2",
        "15.1.0",
        "15.2.0",
        "15.3.0",
        "15.3.1",
        "15.3.2",
        "15.3.3",
        "15.3.4",
        "16.0.0",
        "16.0.0.0b1",
        "16.0.0.0rc1",
        "16.0.0.0rc2",
        "16.1.0",
        "16.2.0",
        "16.3.0",
        "16.3.1",
        "16.3.2",
        "16.4.0",
        "16.4.1",
        "16.4.2",
        "17.0.0",
        "17.0.0.0rc1",
        "17.0.0.0rc2",
        "17.1.0",
        "17.1.1",
        "17.1.2",
        "17.2.0",
        "17.2.1",
        "17.3.0",
        "17.4.0",
        "17.4.1",
        "18.0.0",
        "18.0.0.0rc1",
        "18.0.0.0rc2",
        "18.1.0",
        "18.1.1",
        "18.2.0",
        "18.3.0",
        "18.4.0",
        "18.5.0",
        "18.6.0",
        "19.0.0",
        "19.0.0.0rc1",
        "19.0.0.0rc2",
        "19.1.0",
        "19.2.0",
        "19.3.0",
        "19.4.0",
        "19.5.0",
        "19.6.0",
        "19.7.0",
        "20.0.0",
        "20.0.0.0rc1",
        "20.0.0.0rc2",
        "20.1.0",
        "20.2.0",
        "20.3.0",
        "21.0.0",
        "21.0.0.0rc1",
        "21.0.0.0rc2",
        "21.1.0",
        "22.0.0",
        "22.0.0.0rc1",
        "22.0.0.0rc2"
      ],
      "database_specific": {
        "source": "https://github.com/github/advisory-database/blob/main/advisories/github-reviewed/2022/05/GHSA-4pmp-38hf-rmwj/GHSA-4pmp-38hf-rmwj.json"
      }
    },
    {
      "package": {
        "name": "neutron",
        "ecosystem": "PyPI",
        "purl": "pkg:pypi/neutron"
      },
      "ranges": [
        {
          "type": "ECOSYSTEM",
          "events": [
            {
              "introduced": "2014.1.0"
            },
            {
              "fixed": "2014.1.2"
            }
          ]
        }
      ],
      "database_specific": {
        "source": "https://github.com/github/advisory-database/blob/main/advisories/github-reviewed/2022/05/GHSA-4pmp-38hf-rmwj/GHSA-4pmp-38hf-rmwj.json"
      }
    }
  ],
  "schema_version": "1.4.0"
}
```
